### DISCUSSION TITLE
SpatieMediaLibraryFileUpload reorderable in reverse order

### DISCUSSION THREAD

**Discussion by angus-mcritchie** [filamentphp/filament#11173] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.19

### How can we help you?

I have this in my `ProductResource->form()` method.

```php
$form->schema([
    SpatieMediaLibraryFileUpload::make('images')
        ->reorderable()
        ->multiple()
  ])
```

If save the resource with the media sorted as below.

<img width="635" alt="image" src="https://github.com/filamentphp/filament/assets/53469513/56640e9c-c93f-4713-bd78-dca8f978c051">

The red image `media.order_column` = 1, and the green image `media.order_column` = 2.

I'd expect the order of the first media item to be in the reverse order as the green image is shown first.

Is this the expected behaviour, and is there something I can do to reverse the order?

I'm pretty new to Filament, so it's likely something I'm doing wrong.

Thanks in advance!

**Comment by angus-mcritchie** [comment]
Anyone with the same issue can use `reorderUploadedFilesUsing` on the `Filament\Forms\Components\SpatieMediaLibraryFileUpload` class. Reverse it with this:

```php
->reorderUploadedFilesUsing(function (array $state) {

    $uuids = array_filter(array_values($state));
    $uuids = array_reverse($uuids);  // Only change from vendor/spatie/laravel-medialibrary/src/MediaCollections/Models/Concerns/IsSorted.php:34
    $mediaClass = config('media-library.media_model', Media::class);

    $mappedIds = $mediaClass::query()->whereIn('uuid', $uuids)->pluck('id', 'uuid')->toArray();

    $mediaClass::setNewOrder([
        ...array_flip($uuids),
        ...$mappedIds,
    ]);

    return $state;
})
```

**Reply by javierpomachagua** [reply]
Thanks!

**Comment by munezaclovis** [comment, marked as answer]
I actually found out filepond appends new files at the begining by default. so to have the files be appended at the end(natural order),
you need to call the ->appendFiles()

```php
Forms\Components\SpatieMediaLibraryFileUpload::make('media')
    ->appendFiles()
```

**Reply by saade** [reply]
Lifesaver! I was losing my mind over this.
